### PR TITLE
Create Zone10

### PR DESCRIPTION
**Pull request by matsduf**
Create a new test case that will verify that all servers for a zone
returns exactly one SOA record.

This solves issue #666.

When this PR has been merged an issue will be created in Zonemaster-Engine.

### CODE REVIEW

File: docs/specifications/tests/Zone-TP/zone10.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+
+## Ordered description of steps to be taken to execute the test case
+
+1. Obtain the set of name server IP addresses using [Method4] and [Method5]
+   ("Name Server IP").
```
Comment (sandoche2k) [inline docs/specifications/tests/Zone-TP/zone10.md]: NS IP (Harmonization)

File: docs/specifications/tests/Zone-TP/zone10.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+2. Create a SOA query for the apex of the *Child Zone* with RD flag unset.
+
+3. For each name server in *Name Server IP* do:
```
Comment (sandoche2k) [inline docs/specifications/tests/Zone-TP/zone10.md]: NS IP

File: docs/specifications/tests/Zone-TP/zone10.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+   1. Send the SOA query over UDP to the name server.
+   2. If the name server does not respond with a DNS response, then
+      output *[NO_RESPONSE]*.
+   3. Else, if the DNS response does not include a SOA record in the
+      answer section, then output *[NO_RESPONSE_SOA_QUERY]*.
+   4. Else, if the SOA record or records in the answer section has
+      an owner name same as *Child Zone*, then output
+      *[WRONG_SOA]*.
+   4. Else, if the DNS response includes multipel SOA records in the
```
Comment (sandoche2k) [inline docs/specifications/tests/Zone-TP/zone10.md]: multiple

File: docs/specifications/tests/Zone-TP/zone10.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      answer section, then output *[MULTIPLE_SOA]*.
+
+4. If no message is outputted for any server, then output *[ONE_SOA]*.
```
Comment (sandoche2k) [inline docs/specifications/tests/Zone-TP/zone10.md]: Not clear what does ONE_SOA is

File: docs/specifications/tests/Zone-TP/zone10.md (hunk 1 above)
Comment (matsduf) [reply docs/specifications/tests/Zone-TP/zone10.md]: Updated.

File: docs/specifications/tests/Zone-TP/zone10.md (hunk 2 above)
Comment (matsduf) [reply docs/specifications/tests/Zone-TP/zone10.md]: Updated.

File: docs/specifications/tests/Zone-TP/zone10.md (hunk 3 above)
Comment (matsduf) [reply docs/specifications/tests/Zone-TP/zone10.md]: Updated.

File: docs/specifications/tests/Zone-TP/zone10.md (hunk 4 above)
Comment (matsduf) [reply docs/specifications/tests/Zone-TP/zone10.md]: The message means that all servers have exactly one SOA record for the zone.

File: docs/specifications/tests/Zone-TP/zone10.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+2. Create a SOA query for the apex of the *Child Zone* with RD flag unset.
+
+3. For each name server in *NS IP* do:
+   1. Send the SOA query over UDP to the name server.
+   2. If the name server does not respond with a DNS response, then
+      output *[NO_RESPONSE]*.
+   3. Else, if the DNS response does not include a SOA record in the
+      answer section, then output *[NO_RESPONSE_SOA_QUERY]*.
+   4. Else, if the SOA record or records in the answer section has
+      an owner name same as *Child Zone*, then output
```
[the remark this answers is not in the document]
Comment (matsduf) [reply docs/specifications/tests/Zone-TP/zone10.md]: Updated text.

**Comment by matsduf** [comment]
@sandoche2k, can you approve?

**Comment by matsduf** [comment]
Issue zonemaster/zonemaster-engine#595 has been created for the implementation of the new test case.

**Pull request merged**